sdk/textanalytics/Azure.AI.TextAnalytics/tests/TextAnalyticsClientLiveTests.cs

```diff
@@ -1000,6 +1000,29 @@ public async Task RecognizeLinkedEntitiesBatchWithStatisticsTest()
             Assert.IsNotNull(results.Statistics.ValidDocumentCount);
         }
 
+        [Test]
+        public void RecognizeLinkedEntitiesBatchWithNullIdTest()
+        {
+            TextAnalyticsClient client = GetClient();
+            var documents = new List<TextDocumentInput> { new TextDocumentInput(null, "Hello world") };
+
+            RequestFailedException ex = Assert.ThrowsAsync<RequestFailedException>(async () => await client.RecognizeLinkedEntitiesBatchAsync(documents));
+            Assert.AreEqual(TextAnalyticsErrorCode.InvalidDocument, ex.ErrorCode);
+        }
+
+        [Test]
+        public async Task RecognizeLinkedEntitiesBatchWithNullTextTest()
+        {
+            TextAnalyticsClient client = GetClient();
+            var documents = new List<TextDocumentInput> { new TextDocumentInput("1", null) };
+
+            RecognizeLinkedEntitiesResultCollection results = await client.RecognizeLinkedEntitiesBatchAsync(documents);
+            var exceptionMessage = "Cannot access result for document 1, due to error InvalidDocument: Document text is empty.";
```

Comment:
I believe we discussed about avoiding asserting error messages, since changing them is not breaking. Is there something like an error code you could use? Or do you think that using the message is worth in this case?

Reply:
hmm good point. Because we are constructing the message in the SDK, I wanted to validate it. See here: https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/textanalytics/Azure.AI.TextAnalytics/src/RecognizeLinkedEntitiesResult.cs#L35

I could Assert on `Assert.IsNotNull(ex.Message);`

Reply:
I see. It's in our jurisdiction, so I think it's ok. We could think about something else if we had lots of exception messages do validate in the SDK, but that's not the case.

Reply:
will leave as is for now. If, like you said, it stops being the only custom error I will change.
Thanks!